contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -34,6 +35,24 @@ type htlcIncomingContestResolver struct {
 	htlcSuccessResolver
 }
 
+// newIncomingContestResolver instantiates a new incoming htlc contest resolver.
+func newIncomingContestResolver(htlcExpiry uint32,
```

Comment:
should decode methods just call the constructors also now?

Reply:
hmm idk. it would add boiler plate to first decode everything in local vars. I am already happy that everything is in the resolver file now